s3client.go

```diff
@@ -189,6 +191,48 @@ func (client *s3client) BucketFileVersions(bucketName string, remotePath string)
 	return versions, nil
 }
 
+type BucketListChunk struct {
+	Truncated         bool
+	ContinuationToken *string
+	CommonPrefixes    []string
+	Paths             []string
+}
+
+/* List the S3 bucket `bucketName` content's under `prefix` one chunk at a time
+ *
+ * The returned `BucketListChunk` contains part of the files and subdirectories
+ * present in `bucketName` under `prefix`. The files are listed in `Paths` and
+ * the subdirectories in `CommonPrefixes`. If the returned chunk does not
+ * include all the files and subdirectories, the `Truncated` flag will be set
+ * to `true` and the `ContinuationToken` can be used to retrieve the next chunk.
+ */
+func (client *s3client) ChunkedBucketList(bucketName string, prefix string, continuationToken *string) (BucketListChunk, error) {
+	params := &s3.ListObjectsV2Input{
+		Bucket:            aws.String(bucketName),
+		ContinuationToken: continuationToken,
+		Delimiter:         aws.String("/"),
+		Prefix:            aws.String(prefix),
+	}
+	response, err := client.client.ListObjectsV2(params)
```

Comment:
there is still `ListObjects` in `getBucketContents`, is it possible to replace it to v2? would it bring backward compatibility issue?

Reply:
I would say yes and I don't expect backward compatibility issue. See my answer to your next question for more details as they are related.